From 9c579205669cce54944e2c4f115e69ef18475bbe Mon Sep 17 00:00:00 2001
From: Tycho Andersen <tycho@tycho.ws>
Date: Tue, 23 Jul 2019 09:40:14 -0600
Subject: [PATCH] pidfds: don't print a scary warning on ENOSYS

Most kernels don't have this functionality yet, and so the warning is
printed a lot. Our people are scared of warnings, so let's make it INFO
instead in this case.

Signed-off-by: Tycho Andersen <tycho@tycho.ws>
---
 src/lxc/start.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git src/lxc/start.c src/lxc/start.c
index e3f32f4cb8..e6544ea19c 100644
--- src/lxc/start.c
+++ src/lxc/start.c
@@ -1641,7 +1641,10 @@ static int proc_pidfd_open(pid_t pid)
 
 	/* Test whether we can send signals. */
 	if (lxc_raw_pidfd_send_signal(proc_pidfd, 0, NULL, 0)) {
-		SYSERROR("Failed to send signal through pidfd");
+		if (errno != ENOSYS)
+			SYSERROR("Failed to send signal through pidfd");
+		else
+			INFO("Sending signals through pidfds not supported on this kernel");
 		return -1;
 	}
 
